### PR TITLE
added environment-dev.yaml file

### LINKED ISSUE #699: Create an `environment-dev.yaml` file
We could do with a conda equivalent of `requirements-dev.txt`.

### PR DESCRIPTION
**Pull request by SutarPrasad**
* **Summary of changes** (Bug fix, feature, docs update, ...)
Added environment-dev.yaml file. Updated one documentation.

* **Please check if the PR fulfills these requirements**

- [x] Closes #699 #610 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [ ] Documentation and tutorials updated/added
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [ ] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

### CODE REVIEW

**Comment by SutarPrasad** [comment]
demo output
image

**Review by brendan-m-murphy — CHANGES_REQUESTED** [review]
Looks good, just needs an entry in the change log.

**Review by gareth-j — CHANGES_REQUESTED** [review]
Just need to add in `openghg`'s dependencies

File: environment-dev.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: openghg_dev_env
+channels:
+  - conda-forge
+  - defaults
+  - anaconda
+dependencies:
```
[the remark this answers is not in the document]
Comment (SutarPrasad) [reply environment-dev.yaml:6]: done

File: environment-dev.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  - git
+  - flake8
+  - pre-commit
+  - pytest
+  - pytest-cov
+  - pytest-mock
+  - requests-mock
+  - pytest-tornasync
+  - mypy==1.4.0
+  - types-paramiko
```
[the remark this answers is not in the document]
Comment (SutarPrasad) [reply environment-dev.yaml]: done

**Review by brendan-m-murphy — APPROVED** [review]
Looks good!

**Review by brendan-m-murphy — APPROVED** [review]
Looks like nothing changed from the last review I gave, so still happy to approve this.

**Review by brendan-m-murphy — DISMISSED** [review]
Okay, looks good

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased](https://github.com/openghg/openghg/compare/0.6.2...HEAD)
 
 ### Added
+- Added environment-dev.yaml file for cloud developers environment, especially dependencies like pytest, pytest-mock requested by users. - [PR #769](https://github.com/openghg/openghg/pull/769)
```
Comment (gareth-j) [inline CHANGELOG.md:11]:
```suggestion
- Added `environment-dev.yaml` file for developer conda environment - [PR #769](https://github.com/openghg/openghg/pull/769)
```

**Review by gareth-j — APPROVED** [review]

**Review by brendan-m-murphy — APPROVED** [review]
Looks good!

**Pull request merged**